src/uri_builder.cpp

```diff
@@ -111,7 +111,7 @@ uri_builder &uri_builder::clear_path() {
   return *this;
 }
 
-void uri_builder::append_query(string_type query) {
+void uri_builder::append_query_parameter(string_type query) {
```

Comment:
Consistent naming: `query` -> `name`?

Should make use of `encode_query_component` instead of `encode_query`.